### ISSUE TITLE
SUSE branding: Stylesheets produce invalid HTML 4.01 Transitional [sf#34]

### ISSUE THREAD

**Issue by tomschr** [openSUSE/daps#34]
**Reported by tom_schr on 2012-02-06 16:03 UTC**
(Original from https://bugzilla.novell.com/show_bug.cgi?id=708637):

»Generate a htmlsingle document and check it with http://validator.w3.org/check.«

The generated HTML output is pretty awkward as some obscure HTML constructs are being used. DocBook XSL provides these HTML output formats:
- HTML: this is the old format
- XHTML: these stylesheets are automatically generated from the HTML. They create XHTML 1.0 Transitional files
- XHTML 1.1: same as XHTML, but generates XHTML 1.1 files
- (X)HTML5: New, but partly autogenerated and based on the XHTML stylesheets. Creates HTML5 files.

XHTML, XHTML 1.1, and (X)HTML5 can be improved by a cleanup stylesheet, if needed. Such a solution was implemented in my DoCookBook project (see https://sourceforge.net/p/doccookbook/code/ci/1d81f4c4b7fe5be07e26375afd577afea721eff9/tree/frameworks/xslt/misc/html2cleanhtml.xsl?format=raw).

Although we talked about this, maybe we should reiterate it which option is feasible and useful (also in the light of future things like XSLT2.0).

**Comment by tomschr** [comment]
**Commented by tom_schr on 2012-02-06 16:20**
- Description has changed:

Diff:

```

--- old 
+++ new 
@@ -11,4 +11,4 @@

 XHTML, XHTML 1.1, and (X)HTML5 can be improved by a cleanup stylesheet, if needed. Such a solution was implemented in my DoCookBook project (see <https://sourceforge.net/p/doccookbook/code/ci/1d81f4c4b7fe5be07e26375afd577afea721eff9/tree/frameworks/xslt/misc/html2cleanhtml.xsl?format=raw>).

-Need to talk to Frank, which option is feasible and useful.
+Although we talked about this, maybe we should reiterate it which option is feasible and useful (also in the light of future things like XSLT2.0).

```
- **status**: pending --> needinfo
- **info_provider**: Thomas Schraitle --> Frank Sundermeyer

**Comment by tomschr** [comment]
**Updated by fsundermeyer on 2012-02-08 08:25**
- **status**: needinfo --> open
- **info_provider**: Frank Sundermeyer -->  nobody 
- **priority**:  --> high

**Comment by tomschr** [comment]
**Commented by fsundermeyer on 2012-02-08 08:25**
The current daps version in SVN creates XHTML 1.0 by default and HTML 4.01 an request.

I would suggest to switch to a XHTML 1.1 default. If this produces invalid XHTML, I suggest to provide a "patch" with the SUSE stylesheets (a cleanup stylesheet is probably not needed) and try to get it upstream ASAP.
Keep in mind that version 1.77 of the docbook-xsl-stylesheets is about to be released soon.

I do not think it's necessary to invest time in providing fixes for a valid HTML 4.01 output ATM. I would only do this if we have proof that the 4.01 output is really used.

**Comment by tomschr** [comment]
**Commented by tom_schr on 2012-04-11 12:41**
- Revision 1151 contains a small fix to remove the alt attribute from an &lt;a> tag.
- Revision 1153 changed the &lt;a> "anchor" behaviour to provide a mechanism to _add_ nodes inside an &lt;a> element instead of just append it. This is necessary to avoid constructs of &lt;a id="foo"/>&lt;img .../> which is not valid inside a &lt;pre> element.
- **status**: open --> closed

**Comment by tomschr** [comment]
**Commented by tom_schr on 2012-04-12 06:18**
Addendum:
The above fix creates valid XHTML against version 1.0 Transitional (tested with the DAPS User Guide and Quick Start). 

However, there may be situations which are overly complicated in regards to create valid output. Some constructs in XSLT 1.0 are not very sufficient. In that case, we should think about a "clean up" step with tidy.

**Comment by tomschr** [comment]
**Commented by tom_schr on 2012-04-12 06:31**
Added [Ticket#68](https://sourceforge.net/p/daps/tickets/68/) about clean up step to create valid XHTML output.
